jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -2612,7 +2612,7 @@ public ClassLoader getClassLoader()
                     else
                         callerLoader = callerLoader.getParent();
                 }
-                AccessController.checkPermission(new RuntimePermission("getClassLoader"));
+                System.getSecurityManager().checkPermission(new RuntimePermission("getClassLoader"));
```

Comment:
Shouldn't we remove the member variable `_usingSecurityManager` and replace it by `System.getSecurityManager() != null`?

The value of the variable is set on object creation, but a security manager can be added also after this object has been created.

So, shouldn't we use something like this:

```java
final SecurityManager securityManager = System.getSecurityManager():
if (securityManager == null)
{
    return _classLoader;
} else {
    ...
    securityManager.checkPermission(new RuntimePermission("getClassLoader"));
    return _classloader;
}
```

Makes it sense to provide the `setUsingSecurityManager(boolean usingSecurityManager)` method if `System.getSecurityManager()` returns `null`?

We could also change it to check if `_usingSecurityManager` is true and if it is true, we check if `System.getSecurityManager()` returns a non null value and use that reference, if it returns null, use `AccessControll.checkPermission`

Reply:
Fair point. However I don't really want to remove the getters and setters for _usingSecurityManager from jetty-9.4.  But, in jetty-10 I think it's fine, so we could revert to calling System.getSecurityManager() whenever we need to.

@gregw you introduced the _usingSecurityManager with commit https://github.com/eclipse/jetty.project/commit/d44ba7e24a86d3f16d275b151f1e7fc9755b1ad3
was there any particular reason for doing that rather than calling System.getSecurityManager() when needed? Any reason I wouldn't change it back?

Reply:
I do like the option of not even considering the securityManager - we have been run in environments with very weird securityManager implementations.  So I think that is why we made checking it optional.

Note that if it is set to true and `System.getSecurityManager()` is null, then perhaps that should be an ISE?

But I'm also OK with trialing in jetty-10 of just removing the boolean and only using the existence of the security manager... we will have some alpha and beta releases to find any issues.